### PR TITLE
Add tests to assert current behavior of large future sizes

### PR DESCRIPTION
**Pull request by bryangarza**
Based on a couple of sources:
- https://swatinem.de/blog/future-size/
- https://github.com/rust-lang/rust/issues/62958

### CODE REVIEW

File: tests/ui/future-sizes/large-arg.rs [hunk 1]
```diff
@@ -0,0 +1,17 @@
+// compile-flags: -Z print-type-sizes --crate-type=lib
+// edition: 2021
+// build-pass
```
Comment (compiler-errors) [inline tests/ui/future-sizes/large-arg.rs:3]: The `// ignore-pass` is necessary to skip a "check" build, which causes the UI test failure.

```suggestion
// build-pass
// ignore-pass
```

Comment (bryangarza) [reply tests/ui/future-sizes/large-arg.rs:3]: thanks yeah I saw it earlier when I copied one of the tests over but didn’t quite understand what it’s purpose was (I did see a small explanation in the Rustc Dev Guide)

but now I get what it’s for, thanks :)

File: tests/ui/future-sizes/future-as-arg.rs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+fn main() {
+    let n = std::mem::size_of_val(&use_future(use_future(use_future(use_future(use_future(
+            use_future(use_future(use_future(use_future(use_future(test(
+                [0; 8192]
+            ))))))
+        ))))));
+    println!("size is: {n}");
+    // Not putting an exact number in case it slightly changes over different commits
+    assert!(n > 8000000);
```
Comment (oli-obk) [inline tests/ui/future-sizes/future-as-arg.rs:18]: you can instead (of the println) make this assert print the exact size when it fails

Comment (bryangarza) [reply tests/ui/future-sizes/future-as-arg.rs:18]: oh, yes that's the behavior that I want! I'll do that

**Comment by GuillaumeGomez** [comment]
This test failed in https://github.com/rust-lang/rust/pull/107530.

@bors r-

**Comment by bryangarza** [comment]
I made the test have less nesting and a smaller array, hopefully it can pass the build now

**Comment by bryangarza** [comment]
> in the future, @bryangarza, can you please squash your commits? not sure if it's useful to have 5 commits in the history to add just a couple of UI tests 😂

sounds good! I'll keep that in mind for next time :)

**Pull request merged**